src/content/blog/ddev-backups.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Backing Up DDEV Projects"
+pubDate: 2024-08-10
+#modifiedDate: 2024-04-20
+summary: Backing Up DDEV Projects
+author: Randy Fay
+featureImage:
+  src: /img/blog/2024/08/elephant-walking-backwards.png
+  alt: Elephant walking backwards, 'backing up'
+  credit: 'ChatGPT: An elephant walking backward, with a sign saying "backing up"'
+categories:
+  - DevOps
+---
+
+People sometimes ask how they should back up local DDEV projects. We can do it easily, as most projects are just files in your project directory. Databases have just a bit more work to do.
+
+PHP web projects typically consist of three or four components:
+
+* One or more databases
+* the PHP code
+* User-generated and other non-code transient files (images, etc)
+* Configuration
+
+We need a backup strategy for each of these. 
+
+**Choose a trusted local backup solution.** Time Machine on macOS can work wonderfully for many people, and there are lots of other solutions out there. However, databases have not traditionally been good with backups because they are often not consistent during the backup process, so we need to do something different with them.
+
+**But try not to rely on the local computer anyway.** I try to push code and config to my Git provider (GitHub, GitLab, etc.) and make sure I know how to recreate any user-generated files. Make sure your database contents are accessible from upstream sources like production, or keep good snapshots or exports.
+
+**Databases** need an extra step for safety. In general, don't do work on a database that cannot be recreated by code (with migrations, for example). But most of us want to have a database quickly available as a good place to start from. Since databases are typically in a binary format that can't reliably be backed up, we need a good way to make a copy. DDEV has two great ways to turn databases into files, `ddev snapshot` and `ddev export-db`.
```

Comment:
```suggestion
**Databases** need an extra step for safety. In general, avoid working on a database that cannot be recreated by code (with migrations, for example). However, most of us want to have a database quickly available as a good place to start from. Since databases are typically in a binary format that can't reliably be backed up, we need a good way to make a copy. DDEV has two great ways to turn databases into files, `ddev snapshot` and `ddev export-db`.
```